src/states/client.rs

```diff
@@ -80,30 +73,15 @@ impl Client {
         outbox: &mut EventBox,
     ) -> Transition {
         match routing_msg.content {
-            MessageContent::UserMessagePart {
-                hash,
-                part_count,
-                part_index,
-                payload,
-                ..
-            } => {
+            MessageContent::UserMessage { content, .. } => {
                 trace!(
-                    "{} Got UserMessagePart {:02x}{:02x}{:02x}.., {}/{} from {:?} to {:?}.",
+                    "{} Got UserMessage {:?} from {:?} to {:?}.",
                     self,
-                    hash[0],
-                    hash[1],
-                    hash[2],
-                    part_index + 1,
-                    part_count,
+                    content,
```

Comment:
Would it be OK to omit the content entirely? Otherwise I'd have to provide custom `Debug` implementation for `Request` and `Response` which seems like too much work for little benefit.

Reply:
A digest (hash) of the content shall be enough

Reply:
Without message chunking the digest is useless, so it was removed. I would have to recalculate it which doesn't seem to be worth the performance hit (even though it can be done only when `trace` level is enabled). More importantly, it's not useful. Previously you could use it to verify the integrity of the message after being reassembled from parts and also to know which message parts belong to the same message. None of that applies anymore.

That said, what about outputting the `MessageId` and maybe also whether the message is `Request` or `Response`?

Reply:
Yeah, `MessageId` will be useful. Not sure about `Request` or `Response`.
up to you to decide then.